packages/@webex/plugin-meetings/src/multistream/remoteMediaGroup.ts

```diff
@@ -0,0 +1,222 @@
+   *
+   * @returns {Array<RemoteMedia>}
+   */
+  public getPinnedRemoteMedia() {
+    // return a shallow copy so that the client cannot modify this.pinnedRemoteMedia array
+    return [...this.pinnedRemoteMedia];
+  }
+
+  /**
+   * Pins a specific remote media instance to a specfic CSI, so the media will
+   * no longer come from active speaker, but from that CSI.
+   * If no CSI is given, the current CSI value is used.
+   *
+   */
+  public pin(remoteMedia: RemoteMedia, csi?: CSI): void {
+    // if csi is not specified, use the current one
+    const targetCsi = csi || remoteMedia.csi;
+
+    if (!targetCsi) {
+      throw new Error(
+        `failed to pin a remote media object ${remoteMedia.id}, because it has no CSI set and no CSI value was given`
+      );
+    }
+
+    if (this.pinnedRemoteMedia.indexOf(remoteMedia) >= 0) {
+      if (targetCsi === remoteMedia.csi) {
+        // remote media already pinned to target CSI, nothing to do
+        LoggerProxy.logger.log(
+          `RemoteMediaGroup#pin --> remote media ${remoteMedia.id} already pinned`
+        );
+
+        return;
+      }
+    } else {
+      const idx = this.remoteMedia.indexOf(remoteMedia);
+
+      if (idx < 0) {
+        throw new Error(
+          `failed to pin a remote media object ${remoteMedia.id}, because it is not found in this remote media group`
+        );
+      }
+
+      this.remoteMedia.splice(idx, 1);
+      this.pinnedRemoteMedia.push(remoteMedia);
+
+      this.cancelActiveSpeakerMediaRequest(false);
+      this.sendActiveSpeakerMediaRequest(false);
+    }
+
+    remoteMedia.sendMediaRequest(targetCsi, false);
+    this.mediaRequestManager.commit();
+  }
+
+  /**
+   * Unpins a remote media instance, so that it will again provide media from active speakers
+   *
+   */
+  public unpin(remoteMedia: RemoteMedia) {
+    if (this.remoteMedia.indexOf(remoteMedia) >= 0) {
+      LoggerProxy.logger.log(
+        `RemoteMediaGroup#pin --> remote media ${remoteMedia.id} already unpinned`
+      );
+
+      return;
+    }
+    const idx = this.pinnedRemoteMedia.indexOf(remoteMedia);
+
+    if (idx < 0) {
+      throw new Error(
+        `failed to unpin a remote media object ${remoteMedia.id}, because it is not found in this remote media group`
+      );
+    }
+
+    this.pinnedRemoteMedia.splice(idx, 1);
+    this.remoteMedia.push(remoteMedia);
+
+    remoteMedia.cancelMediaRequest(false);
+    this.cancelActiveSpeakerMediaRequest(false);
+    this.sendActiveSpeakerMediaRequest(false);
+    this.mediaRequestManager.commit();
+  }
+
+  public isPinned(remoteMedia: RemoteMedia) {
+    if (this.remoteMedia.indexOf(remoteMedia) >= 0) {
+      return false;
+    }
+    if (this.pinnedRemoteMedia.indexOf(remoteMedia) >= 0) {
+      return true;
+    }
+
+    throw new Error(`remote media object ${remoteMedia.id} not found in the group`);
+  }
+
+  private sendActiveSpeakerMediaRequest(commit: boolean) {
+    if (this.mediaRequestId) {
```

Comment:
unnecessary check since it's done in cancelActiveSpeakerMediaRequest

Reply:
check removed